packages/core/saved-objects/core-saved-objects-utils-server/src/saved_objects_test_utils.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { isEqual } from 'lodash';
+
+/**
+ * Determines if a given Set is equal to another given Set. Set types must be the same, and comparable.
+ *
+ * @param setA the first Set to compare
+ * @param setB the second Set to compare
+ * @returns {boolean} True if Set A is equal to Set B
+ */
+export function setsAreEqual<T>(setA: Set<T>, setB: Set<T>) {
+  if (setA.size !== setB.size) return false;
+
+  const NotEqualException = {};
+  try {
+    setA.forEach((element) => {
+      // End the loop early if we found an inequality
+      if (!setB.has(element)) throw NotEqualException;
+    });
+  } catch (e) {
+    if (e === NotEqualException) return false;
+    throw e;
+  }
+
+  return true;
+}
+
+/**
+ * Determines if a given map of arrays is equal to another given map of arrays.
+ * Used for comparing namespace maps in saved object repo/security extension tests.
+ *
+ * @param mapA the first map to compare
+ * @param mapB the second map to compare
+ * @returns {boolean} True if map A is equal to map B
+ */
+export function arrayMapsAreEqual<T>(mapA: Map<T, T[] | undefined>, mapB: Map<T, T[] | undefined>) {
```

Comment:
__optional nit:__ (I see it was like that before your PR, so feel free to ignore)
Maybe something like this would be a bit more efficient (cannot say it's super readable though, too bad we have to support both `undefined` and `T[]` for values - or is there an option to support only `T[]`?)?
```ts
  if (mapA.size !== mapB.size) {
    return false;
  }

  for (const [keyA, valueA] of mapA) {
    // Compare keys....
    if (!mapB.has(keyA)) {
      return false;
    }

    // Compare values (bit complex since we need to distinguish undefined vs empty array)....
    const valueB = mapB.get(keyA);
    if (valueA?.length !== valueB?.length || !setsAreEqual(new Set(valueA), new Set(valueB))) {
      return false;
    }
  }

  return true;
```

Reply:
Thanks for the feedback here and on the foreach loop. I worked through the compare functions again and feel they are better now, and at least contain no conversions. I wanted to avoid converting the array to a Set because it inherently alters the data (eliminates duplicates, etc).